config/mysqlpersister/mysqlpersister.go

```diff
@@ -0,0 +1,210 @@
+package mysqlpersister
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/golang/protobuf/proto"
+	"github.com/jmoiron/sqlx"
+
+	"github.com/square/quotaservice/config"
+	"github.com/square/quotaservice/logging"
+	qsc "github.com/square/quotaservice/protos/config"
+)
+
+type mysqlPersister struct {
+	latestVersion int
+	db            *sqlx.DB
+	m             *sync.RWMutex
+
+	watcher  chan struct{}
+	shutdown chan struct{}
+
+	activeFetchers *sync.WaitGroup
+
+	configs map[int]*qsc.ServiceConfig
+}
+
+type configRow struct {
+	Version int    `db:"Version"`
+	Config  string `db:"Config"`
+}
+
+func New(dbUser, dbPass, dbHost string, dbPort int, dbName string, pollingInterval time.Duration) (config.ConfigPersister, error) {
+	db, err := sqlx.Open("mysql",
+		fmt.Sprintf("%s:%s@(%s:%v)/%s",
+			dbUser,
+			dbPass,
+			dbHost,
+			dbPort,
+			dbName))
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = db.Query("SELECT 1 FROM quotaservice LIMIT 1")
+	if err != nil {
+		return nil, errors.New("table quotaservice does not exist")
+	}
+
+	mp := &mysqlPersister{
+		db:             db,
+		configs:        make(map[int]*qsc.ServiceConfig),
+		activeFetchers: &sync.WaitGroup{},
+		m:              &sync.RWMutex{},
+		watcher:        make(chan struct{}),
+		shutdown:       make(chan struct{}),
+		latestVersion:  -1,
+	}
+
+	mp.pullConfigs()
+
+	mp.activeFetchers.Add(1)
```

Comment:
I'm not totally following this. Is it expected that there will be multiple fetchers? If so, under what conditions do we expect that?

Reply:
Good point, this is just a pattern I use a lot that I tend to slip into.  I changed it to be a channel.